server/src/main/java/com/kritsit/casetracker/server/datalayer/DatabasePersistence.java

```diff
@@ -53,11 +53,14 @@ public boolean isOpen() {
 
     private ResultSet get(String sql) throws SQLException {
         logger.info("Executing request {}", sql);
-        ResultSet query = null;
-        Statement statement = connection.createStatement(ResultSet.CONCUR_READ_ONLY, 
-                                                         ResultSet.TYPE_FORWARD_ONLY);
-        query = statement.executeQuery(sql);
-        return query;
+        
+        try(Statement statement = connection.createStatement(ResultSet.CONCUR_READ_ONLY, 
+                ResultSet.TYPE_FORWARD_ONLY)) {
+             ResultSet query = statement.executeQuery(sql);
+             return query;
+        } catch (SQLException e){
+            throw e;
+        }
     }
```

Comment:
"Method may fail to clean up stream or resource on checked exception"
"Method may fail to close database resource"
The statement is not getting closed.

"Nonconstant string passed to execute method on an SQL statement"
I'm not sure what to do about this. Making the incoming statement string final does not clear it. I think it might want it to use prepared statements. This is something that may be worth doing as part of the response to the "input validation" issue.